### PR TITLE
feat(nuxt): add useRouteQuery and useRouteParams

### LINKED ISSUE #24365: useQuery and useParams composables
### Describe the feature

useQuery should return all route queries.
useParams should all return all route params.

Currently we have to first call the useRoute composable before getting the route queries and or route params.

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

### PR DESCRIPTION
**Pull request by saidinusah**
### 🔗 Linked issue
#24365

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

 I've added useRouteQuery and useRouteParams so we can easily get the route query and params respectively

It prevents us from doing useRoute?.query or useRoute?.params so many times in our code base. Instead with useRouteQuery returns the route queries if there are any and undefined if none.
useRouteParams also returns the route params if there are any and undefined in none

Resolves #24365

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by DamianGlowala** [comment]
Could you please lint your work with `pnpm lint:fix` (notice many changes in indentation, added semicolons, etc.)?

File: packages/nuxt/src/app/composables/index.ts [hunk 1]
```diff
@@ -1,36 +1,76 @@
...
 export {
   /** @deprecated Import `useHead` from `#imports` instead. This may be removed in a future minor version. */
   useHead,
   /** @deprecated Import `useSeoMeta` from `#imports` instead. This may be removed in a future minor version. */
   useSeoMeta,
   /** @deprecated Import `useServerSeoMeta` from `#imports` instead. This may be removed in a future minor version. */
-  useServerSeoMeta
-} from '@unhead/vue'
+  useServerSeoMeta,
+} from "@unhead/vue";
```
Comment (manniL) [inline packages/nuxt/src/app/composables/index.ts:13]: Please remove auto-formatting changes (such as this one) from the PR ☺️

**Review by TheAlexLichter — CHANGES_REQUESTED** [review]

File: packages/nuxt/src/app/composables/index.ts (hunk 1 above)
Comment (DamianGlowala) [reply packages/nuxt/src/app/composables/index.ts:13]: Mentioned precisely that just above 😄

Comment (1nusah) [reply packages/nuxt/src/app/composables/index.ts:13]: pnpm lint:fix doesn't fix it so i'm reverting the changes and working on that again. this time with prettier turned off

Comment (1nusah) [reply packages/nuxt/src/app/composables/index.ts]: should be good now

**Review by danielroe — COMMENTED** [review]
I need to think about whether this adds enough DX improvement, as it's a pretty thin wrapper. 

In addition, both of these destructure the params/queries from the reactive route object so they are then non-reactive. (If you call them and then transition route the values will be incorrect.) We could fix that by using a computed value.

**Comment by DamianGlowala** [comment]
As much as the efforts to provide this PR are appreciated, I'd be personally reluctant to merge these for the following reasons:
- I agree these are really thin wrappers (`useRoute().params` vs `useRouteParams()`), as @danielroe mentioned above
- these two can be added with minimal effort to the `composables/` directory in the users' projects (if they believe these add significant value), as mentioned by @manniL
- some might argue that if we provide these two, why not have `useRouteFullPath()`/`useRoutePath()`/`useRouteName()`, etc.

... and instead recommend to the author to add these two in their own projects.

**Comment by pi0** [comment]
Agree with all the points. It _might_ be useful if we support utilities to access a single reactive param/query by name though like `useRouteQuery(name: string): Ref<string>` since it will **solve** a common mistake of losing reactivity by destructing but probably for top level props of `useRoute().*` it is overkill to have core utils for each or some.

**Comment by danielroe** [comment]
@1nusah apologies for your work on this. I think they are good composables but probably shouldn't be built-in to Nuxt in this case.

**Pull request closed without merging**